iOS Client/HTTPClient/MBClient.swift

```diff
@@ -106,6 +108,62 @@ class MBClient: NSObject {
         }.resume()
     }
     
+    func getImageData(imageID: Int, completion: @escaping (UIImage?) -> Void) {
+        let urlString = "\(baseURL)\(mediaEndpoint)/\(imageID)"
+        guard let url = URL(string: urlString) else {
+            completion(nil)
+            return
+        }
+        
+        print("firing get media request: \(imageID)")
+        self.session.dataTask(with: url) { (data: Data?, resp: URLResponse?, err: Error?) in
+            guard self.wasDataTaskSuccessful(resp: resp, err: err) else {
+                completion(nil)
+                return
+            }
+            
+            // Process Data
+            guard let mediaData = data else {
+                completion(nil)
+                return
+            }
+            
+            var json : Any
+            do {
+                json = try JSONSerialization.jsonObject(with: mediaData, options: .allowFragments)
+            } catch let error as NSError {
+                print("Could not get media object. \(error), \(error.userInfo)")
+                completion(nil)
+                return
+            }
+            
+            if let arr = json as? NSDictionary, let imageLink = arr.value(forKeyPath: "media_details.sizes.thumbnail.source_url") as? String {
+                guard let imageUrl = URL(string: imageLink) else {
+                    completion(nil)
+                    return
+                }
+                print("firing get image request: \(imageID)")
+                self.session.dataTask(with: imageUrl) { (data: Data?, resp: URLResponse?, err: Error?) in
+                    guard self.wasDataTaskSuccessful(resp: resp, err: err) else {
+                        completion(nil)
+                        return
+                    }
+                    
+                    // Process Data
+                    guard let imageData = data, let image = UIImage(data: imageData) else {
+                        completion(nil)
+                        return
+                    }
+                    
+                    completion(image)
+                }.resume()
+            } else {
+                completion(nil)
+            }
+            
+        }.resume()
+    }
+    
```

Comment:
I'm wondering if we should have used a library like PromiseKit so we don't get nested completion blocks like this. Might be too late but it has been pretty helpful on Synchrony.